# Kiểm định giả thiết thống kê và ý nghĩa của trị số P (P-value)

## 7.1 Trị số P
Trong nghiên cứu khoa học, ngoài những dữ kiện bằng số, biểu đồ và hình ảnh,
con số mà chúng ta thường hay gặp nhất là trị số P (mà tiếng Anh gọi là P-value). Trong
các chương sau đây, bạn đọc sẽ gặp trị số P rất nhiều lần, và đại đa số các suy luận phân
tích thống kê, suy luận khoa học đều dựa vào trị số P. Do đó, trước khi bàn đến các
phương pháp phân tích thống kê bằng R, tôi thấy cần phải có đôi lời về ý nghĩa của trị số
này.

Trị số P là một con số xác suất, tức là viết tắt chữ “probability value”. Chúng ta
thường gặp những phát biểu được kèm theo con số, chẳng hạn như “Kết quả phân tích
cho thấy tỉ lệ gãy xương trong nhóm bệnh nhân được điều trị bằng thuốc Alendronate là
2%, thấp hơn tỉ lệ trong nhóm bệnh nhân không được chữa trị (5%), và mức độ khác biệt
này có ý nghĩa thống kê (p = 0.01)”, hay một phát biểu như “Sau 3 tháng điều trị, mức độ
giảm áp suất máu trong nhóm bệnh nhân là 10% (p < 0.05)”. Trong văn cảnh trên đây,
đại đa số nhà khoa học hiểu rằng trị số P phản ánh xác suất sự hiệu nghiệm của thuốc
Alendronate hay một thuật điều trị, họ hiểu rằng câu văn trên có nghĩa là “xác suất mà
thuốc Alendronate tốt hơn giả dược là 0.99” (lấy 1 trừ cho 0.01). Nhưng cách hiểu đó
hoàn toàn sai!

Trong “Từ điển toán kinh tế thống kê, kinh tế lượng Anh – Việt” (Nhà xuất bản
Khoa học và Kĩ thuật, 2004), tác giả định nghĩa trị số P như sau: “P – giá trị (hoặc giá
trị xác suất). P giá trị là mức ý nghĩa thống kê thấp nhất mà ở đó giá trị quan sát được
của thống kê kiểm định có ý nghĩa” (trang 690). Định nghĩa này thật là khó hiểu! Thật ra
đó cũng là định nghĩa chung mà các sách khoa Tây phương thường hay viết. Lật bất cứ
sách giáo khoa nào bằng tiếng Anh, chúng ta sẽ thấy một định nghĩa về trị số P na ná
giống nhau như “Trị số P là xác suất mà mức độ khác biệt quan sát do các yếu tố ngẫu
nhiên gây ra (P value is the probability that the observed difference arose by chance)”.
Thật ra định nghĩa này chưa đầy đủ, nếu không muốn nói là … sai. Chính vì sự mù mờ
của định nghĩa cho nên rất nhiều nhà khoa học hiểu sai ý nghĩa của trị số P.
Thật vậy, rất nhiều người, không chỉ người đọc mà ngay cả chính các tác giả của
những bài báo khoa học, không hiểu ý nghĩa của trị số P. Theo một nghiên cứu được
công bố trên tập san danh tiếng Statistics in Medicine [1], tác giả cho biết 85% các tác giả
khoa học và bác sĩ nghiên cứu không hiểu hay hiểu sai ý nghĩa của trị số P. Đọc đến đây
có lẽ bạn đọc rất ngạc nhiên, bởi vì điều này có nghĩa là nhiều nhà nghiên cứu khoa học
có khi không hiểu hay hiểu sai những gì chính họ viết ra có nghĩa gì! Thế thì, câu hỏi cần
đặt ra một cách nghiêm chỉnh: Ý nghĩa của trị số P là gì? Để trả lời cho câu hỏi này, chúng ta cần phải xem xét qua khái niệm phản nghiệm và tiến trình của một nghiên cứu
khoa học.

## 7.2 Giả thiết khoa học và phản nghiệm
Xem trong sách chương 7, trang 2

## 7.3 Ý nghĩa của trị số P qua mô phỏng
Để hiểu ý nghĩa thực tế của trị số P, tôi sẽ nêu một ví dụ đơn giản như sau:
Ví dụ 1. Một thí nghiệm được tiến hành để tìm hiểu sở thích của người tiêu thụ
đối với hai loại cà phê (hãy tạm gọi là cà phê A và B). Các nhà nghiên cứu cho 50 khách
hàng uống thử hai loại cà phê trong cùng một điều kiện, và hỏi họ thích loại cà phê nào.
Kết quả cho thấy 35 người thích cà phê A, và 15 người thích cà phê B. Vấn để đặt ra là
qua kết quả này, các nhà nghiên cứu có thể kết luận rằng cà phê loại A được ưa chuộng
hơn cà phê B, hay kết quả trên chỉ là do ngẫu nhiên mà ra?
“Do ngẫu nhiên mà ra” có nghĩa là theo luật nhị phân, khả năng mà kết quả trên
xảy ra là bao nhiêu? Do đó, lí thuyết xác suất nhị phân có phần ứng dụng trong trường
hợp này, bởi vì kết quả của nghiên cứu chỉ có hai “giá trị” (hoặc là thích A, hoặc thích
B).

Nói theo ngôn ngữ của phản nghiệm, giả thiết đảo là nếu không có sự khác biệt về
sở thích, xác suất mà một khách hàng ưa chuộng một loại cà phê là 0.5. Nếu giả thiết này
là đúng (tức p = 0.5, p ở đây là xác suất thích cà phê A), và nếu nghiên cứu trên được lặp
đi lặp lại (chẳng hạn như) 1000 lần, và mỗi lần vẫn 50 khách hàng, thì có bao nhiêu lần
với 35 khách hàng ưa chuộng cà phê A? Gọi số lần nghiên cứu mà 35 (hay nhiều hơn)
trong số 50 thích cà phê A là “biến cố” X, nói theo ngôn ngữ xác suất, chúng ta muốn tìm
P(X | p=0.50) =?

Để trả lời câu hỏi này, chúng ta có thể ứng dụng hàm rbinom để mô phỏng vì
như nói trên thực chất của vấn đề là một phân phối nhị phân:

In [2]:
bin <- rbinom(1000, 50, 0.5)
bin

[1] 25 31 21 26 23 27 24 29 25 20 21 20 30 28 24 33 25 25 27 23 24 33 19 24
  [25] 19 21 26 20 24 25 24 25 29 24 27 27 28 20 24 28 28 22 30 22 21 28 20 27
  [49] 25 25 26 25 30 25 24 25 19 20 23 29 28 26 30 24 27 31 30 23 24 29 30 23
  [73] 25 22 23 28 26 22 21 34 30 26 25 23 23 17 28 30 28 17 29 20 25 32 22 27
  [97] 27 21 24 28 26 24 26 27 29 31 24 26 21 31 27 24 25 22 34 29 23 21 21 23
 [121] 27 20 23 25 32 23 24 25 25 22 22 27 25 27 27 23 30 26 25 25 23 20 28 25
 [145] 26 23 23 21 29 25 31 30 20 24 28 22 24 26 27 22 29 25 29 24 16 28 22 18
 [169] 29 31 25 20 31 21 27 31 23 25 32 30 21 21 23 15 22 23 25 24 17 22 29 26
 [193] 28 22 33 25 25 22 21 23 32 23 28 23 18 27 28 25 29 24 21 30 23 28 22 23
 [217] 25 26 27 25 24 20 29 29 30 19 26 26 24 29 25 18 29 28 22 21 26 33 22 30
 [241] 26 22 27 24 23 27 16 18 26 25 31 27 20 28 27 25 27 31 25 22 25 23 29 35
 [265] 26 32 23 27 27 26 21 22 26 25 18 24 33 24 32 26 19 27 23 26 31 26 18 26
 [289] 21 20 23 24 27 34 26 23 23 19 31 26 19 26 26 33 21 29 20 23 26 18 19 26
 [313] 29 22 32 25 21 27 29 23 27 27 22 23 19 27 27 25 26 27 20 24 25 26 31 23
 [337] 21 18 24 27 25 24 22 23 23 21 22 25 24 26 26 24 20 25 21 17 26 29 24 28
 [361] 27 24 28 22 19 26 29 22 22 20 25 31 26 28 28 30 22 33 30 30 28 23 20 28
 [385] 21 27 26 27 29 27 26 24 21 21 24 27 22 31 22 22 26 25 23 29 27 21 31 32
 [409] 27 22 25 24 25 19 28 24 20 24 31 23 25 27 32 21 26 20 26 29 20 26 31 22
 [433] 24 28 20 26 27 29 26 24 22 19 30 19 24 27 24 26 26 25 28 19 27 28 27 18
 [457] 20 21 25 24 25 31 27 29 31 33 22 30 29 21 22 22 27 22 25 29 25 27 25 26
 [481] 26 24 21 20 26 30 27 27 21 27 31 32 30 20 33 23 28 26 25 23 28 22 24 22
 [505] 26 22 29 27 24 31 32 26 31 23 30 24 27 24 22 29 26 28 28 29 25 28 24 26
 [529] 20 25 26 20 31 27 33 25 22 26 24 25 33 28 25 29 27 28 22 20 24 28 26 20
 [553] 28 33 28 21 17 27 30 30 23 26 15 24 30 26 26 25 25 22 24 36 28 26 19 24
 [577] 26 23 27 17 33 21 29 25 21 28 29 28 23 25 22 32 27 24 22 27 33 21 27 16
 [601] 22 31 25 28 21 19 22 23 26 27 28 24 17 33 28 27 26 27 27 27 22 26 27 26
 [625] 23 34 30 26 24 21 27 21 24 28 23 31 20 24 20 26 30 23 20 23 21 27 24 24
 [649] 26 28 21 23 28 21 30 23 27 20 23 20 22 26 30 23 24 25 27 29 30 27 23 21
 [673] 26 29 22 32 20 25 25 22 23 22 25 31 22 28 30 19 24 24 29 21 27 29 27 24
 [697] 17 23 22 26 31 31 29 21 30 30 26 22 18 24 26 22 21 26 27 27 23 23 23 27
 [721] 22 23 28 27 21 30 24 24 24 22 28 26 28 32 25 25 26 31 22 28 22 19 25 17
 [745] 25 22 28 28 30 29 27 28 24 26 24 25 28 26 29 26 32 29 23 28 24 26 26 26
 [769] 30 29 25 28 24 23 20 29 28 21 23 25 25 25 24 30 31 23 23 24 27 24 23 25
 [793] 22 24 36 25 28 27 24 32 30 28 24 27 28 23 21 35 23 22 19 25 29 24 30 29
 [817] 24 24 19 27 29 22 27 29 26 24 19 23 22 22 21 31 22 22 25 25 27 21 20 22
 [841] 29 27 28 29 25 23 24 28 24 28 25 23 22 25 23 22 24 25 24 26 25 25 20 25
 [865] 24 22 25 24 24 26 26 31 21 29 36 27 23 23 28 22 28 28 26 25 19 22 20 23
 [889] 22 21 27 25 23 25 26 30 26 22 30 23 20 26 27 19 21 26 32 15 22 25 25 24
 [913] 31 25 23 24 20 27 22 24 22 23 28 27 21 21 24 24 22 24 25 28 18 19 25 29
 [937] 23 26 25 27 28 27 23 28 20 26 27 28 29 21 26 27 26 31 22 29 26 22 26 25
 [961] 18 30 27 31 22 23 30 25 21 27 18 31 32 28 22 33 27 28 27 28 27 30 26 24
 [985] 27 31 27 29 23 24 21 25 27 24 19 25 24 21 27 26

Trong lệnh trên, chúng ta yêu cầu R mô phỏng 1000 lần nghiên cứu, mỗi lần có 50 khách
hàng, và theo giả thiết đảo, xác suất thích A là 0.50. Để biết kết quả của mô phỏng đó,
chúng ta sử dụng hàm table như sau:

In [3]:
table(bin)

bin
 15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34 
  3   3   9  13  25  43  60  87  85 101 105 103 104  77  57  44  38  18  16   4 
 35  36 
  2   3 

Qua kết quả trên, chúng ta thấy trong số 1000 “nghiên cứu” đó, chỉ có 3 nghiên cứu mà
số khách hàng thích cà phê A là 35 người (với điều kiện không có khác biệt giữa hai loại
cà phê, hay nói đúng hơn là nếu p =0.5). Nói cách khác:
$P(X ≥ 35 | p=0.50) = 3/1000 = 0.003$

Chúng ta cũng có thể thể hiện tần số trên bằng một biểu đồ tần số như sau:

Tất nhiên chúng ta có thể làm một mô phỏng khác với số lần tái thí nghiệm là
100.000 lần (thay vì 1000 lần) và tính xác suất $P(X ≥ 35 | p=0.50)$.

In [5]:
bin <- rbinom(100000, 50, 0.5)
bin <- rbinom(100000, 50, 0.5)
table(bin)

bin
   10    11    12    13    14    15    16    17    18    19    20    21    22 
    3     6    10    36    74   197   391   894  1639  2699  4218  5856  8084 
   23    24    25    26    27    28    29    30    31    32    33    34    35 
 9567 10903 11235 10649  9640  7789  6014  4197  2649  1614   859   441   200 
   36    37    38    39    40 
   83    35     9     6     3 

Lần này, chúng ta có nhiều khả năng hơn (vì số lần mô phỏng tăng lên). Chẳng hạn như
có thể có nghiên cứu cho ra 11 khách hàng (tối thiểu) hay 40 khách hàng (tối đa) thích cà phê A. Nhưng chúng ta muốn biết số lần nghiên cứu mà 35 khách hàng trở lên thích cà phê A, và kết quả trên cho chúng ta biết, xác suất đó là:

In [6]:
(223+98+21+5+7+1)/100000

[1] 0.00355

Nói cách khác, xác suất $P(X ≥ 35 | p=0.50)$ quá thấp (chỉ 0.3%), chúng ta có bằng
chứng để cho rằng kết quả trên có thể không do các yếu tố ngẫu nhiên gây nên; tức có
một sự khác biệt về sở thích của khách hàng đối với hai loại cà phê.

Con số P = 0.0035 chính là trị số P. Theo một qui ước khoa học, tất cả các trị số
P thấp hơn 0.05 (tức thấp hơn 5%) được xem là “significant”, tức là “có ý nghĩa thống
kê”.